docs/SETUP_PROVISIONING.md

```diff
@@ -0,0 +1,1505 @@
+in the `~/.ssh/known_hosts` file associated with the elastic IP address if this
+is not the first time you have performed this provisioning step. 
+
+Ansible will update values for the project environment variables below in 
+GitLab. Check them on the project environment variables page after the Ansible
+provisioning has completed. This is done by the `docker-postinstall` role.
+
+| Variable's Key | Environnments | Description |
+|---|---|---|
+| docker_tlsauth_ca | staging | Certificate authority for staging server - this is provided by the staging server during Ansible provisioning |
+| docker_tlsauth_cert | staging | Public certificate for staging server - this is provided by staging server during Ansible provisioning |
+| docker_tlsauth_key  | staging | the server key for the above CA - this is provided by staging server during Ansible provisioning |
+| docker_tlsauth_ca | live | Certificate authority for staging server - this is provided by the live server during Ansible provisioning |
+| docker_tlsauth_cert | live | Public certificate for staging server - this is provided by live server during Ansible provisioning |
+| docker_tlsauth_key  | live | the server key for the above CA - this is provided by live server during Ansible provisioning |
+| remote_public_ip    | staging | Public IP address of staging server |
+| remote_private_ip   | staging | Private IP address of staging server |
+| remote_public_ip    | live | Public IP address of live server |
+| remote_private_ip   | live | Private IP address of live server |
+
+This is for running a secure Docker engine on the AWS EC2 server
+so that the Docker API is secured over TCP and we know we are communicating 
+with the correct server and not a malicious impersonation. We also need to 
+authenticate the client with TLS so only clients using the client certificates 
+can use the Docker engine. This is the 2-way certificate-based authentication.
+
+>When Ansible generates the client/server certificate, it writes them on the EC2 instance at location ``/home/centos/.docker/``
+
+>If an operator needs to perform a docker action on the EC2 instance from this development machine,
+the three files constituting the client certificates `ca.pem`, `cert.pem` and `key.pem` in `ops/infrastructure/envs/<DEPLOY_ENV>/output/` or from gitlab variables need to be copied to ``~/.docker/`` in development machine.
+Then the containers in dockerhost server can be accessed like this:
+```
+docker --tlsverify -H=<remote_public_ip>:2376 ps
+```
+
+The RDS instance is provisioned with a database via the bastion server by a
+separate ansible playbook:
+```
+$ cd ops/infrastructure/envs/staging
+$ ansible-playbook -i ../../inventories bastion_playbook.yml
+```
+
+The bastion playbook will create a `gigadb` database containing data from
+`sql/production_like.pgdmp`. This file is created by running `./up.sh` when
+spinning up a local GigaDB on your development platform.
```

Comment:
I think the bastion playbook will create a database on RDS by restoring from a backup from S3 now.